pkg/apis/pipeline/v1alpha1/verificationpolicy_types.go

```diff
@@ -0,0 +1,94 @@
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +genclient
+// +genclient:noStatus
+// +genreconciler:krshapedlogic=false
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VerificationPolicy defines the resources and corresponding authorities to verify.
+//
+// +k8s:openapi-gen=true
+type VerificationPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec holds the desired state of the VerificationPolicy.
+	Spec VerificationPolicySpec `json:"spec"`
+}
+
+// VerificationPolicyList contains a list of VerificationPolicy
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VerificationPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VerificationPolicy `json:"items"`
+}
+
+// GetGroupVersionKind implements kmeta.OwnerRefable.
+func (*VerificationPolicy) GetGroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind("VerificationPolicy")
+}
+
+// VerificationPolicySpec defines the patterns and authorities.
+type VerificationPolicySpec struct {
+	// Resources defines the patterns of Resources names that should be subject to this policy. For example, we may want to apply this Policy only from a certain github repo. Then the ResourcesPattern should include the path. If using gitresolver, and we want to config keys from a certain git repo. `ResourcesPattern` can be `https://github.com/tektoncd/catalog.git`, we will use regex to filter out those resources.
+	Resources []ResourcePattern `json:"resources"`
+	// Authorities defines the rules for validating signatures.
+	Authorities []Authority `json:"authorities"`
+}
+
+// ResourcePattern defines the pattern of the resource source
+type ResourcePattern struct {
+	// Pattern defines a resource pattern. Regex is created to filter resources based on `Pattern`
+	// Examples patterns:
+	// Github resource: https://github.com/tektoncd/catalog.git, https://github.com/tektoncd/*
+	// Bundle resource: gcr.io/tekton-releases/catalog/upstream/git-clone, gcr.io/tekton-releases/catalog/upstream/*
+	// Hub resource: https://artifacthub.io/*,
+	Pattern string `json:"pattern"`
+}
+
+// The Authority block defines the rules for validating signatures.
+type Authority struct {
+	// Name is the name for this authority.
+	Name string `json:"name"`
+	// Key defines the type of key to validate the resource.
+	Key *KeyRef `json:"key,omitempty"`
+}
+
+// KeyRef defines the reference to a public key
+type KeyRef struct {
+	// SecretRef sets a reference to a secret with the key.
+	// +optional
+	SecretRef *v1.SecretReference `json:"secretRef,omitempty"`
+	// Data contains the inline public key.
+	// +optional
+	Data []string `json:"data,omitempty"`
```

Comment:
Yes this is a great question! It depends on whether we allow multiple keys in `SecretRef`. If it is allowed then we need to have a list of data to store keys. In the current PR we don't enforce to have 1 key in `KeyRef`. But it may be more intuitive to users to only have one key in `KeyRef`.

policy-controller enforces this, so we can also follow the same strategy